SYCL/ESIMD/aot_mixed.cpp

```diff
@@ -0,0 +1,173 @@
+//==---------------- aot_mixed.cpp  - DPC++ ESIMD on-device test -----------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu
+// UNSUPPORTED: cuda || hip
+// RUN: %clangxx -fsycl -fsycl-targets=spir64_gen -Xsycl-target-backend=spir64_gen "-device gen9" -o %t.sycl.out -DENABLE_SYCL=0 %s
+// RUN: %GPU_RUN_PLACEHOLDER %t.sycl.out
+// RUN: %clangxx -fsycl -fsycl-targets=spir64_gen -Xsycl-target-backend=spir64_gen "-device gen9" -o %t.out %s
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// XFAIL: *
+// TODO: remove XFAIL once ocloc support for automatic scalar/vector SPIRV
+// module dispatching is available in public drivers. Also change 'gen9' (safe
+// variant to reliably get unexpected PASS when ocloc is fixed) to '*' (which
+// stresses ocloc).
+
+// This test checks ESIMD ahead-of-time compilation scenarios:
+// 1) When the application contains both SYCL and ESIMD kernel, thus requiring
+//    different GPU back-ends (scalar and vector) to kick-in at compile-time.
+// 2) When the application contains only ESIMD kernel.
+
+#include "esimd_test_utils.hpp"
+
+#include <CL/sycl.hpp>
+#include <iostream>
+#include <sycl/ext/intel/experimental/esimd.hpp>
+
+using namespace cl::sycl;
+
+#ifndef ENABLE_SIMD
```

Comment:
It is unclear here, why this macro is needed? It is just turned on in the next line all the time because it is not set in the compilation line. Is it supposed to be disabled some time later? It may be good to have a TODO comment here.

Reply:
ENABLE_SIMD/ENABLE_SYCL are used to enable/disable corresponding parts in the test. e.g. -DENABLE_SYCL=0 disables the SYCL part, which is used in one of the RUN: commands above.